dex/btc/clone.go

```diff
@@ -0,0 +1,44 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package btc
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+// ReadCloneParams translates a network parameters struct. This entire module
+// may be usable by a BTC clone if certain conditions are met. Many BTC clones
+// have btcd forks with their own Go config files. See also CloneParams.
+func ReadCloneParams(cloneParams interface{}) (*chaincfg.Params, error) {
+	p := new(chaincfg.Params)
+	cloneJSON, err := json.Marshal(cloneParams)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling network params: %v", err)
+	}
+	err = json.Unmarshal(cloneJSON, &p)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling network params: %v", err)
+	}
```

Comment:
passing through Marshall=>Unmarshal certainly does the trick, but I feel like we should look into using reflect here at some point.  Something along the lines of 

```go
package main

import (
	"fmt"
	"reflect"
)

func main() {
	type Tbig struct {
		A int
		B string
	}

	type Tsmall struct {
		B string
	}

	t := &Tbig{42, "dcr"}
	tClone := &Tsmall{"ltc"}

	s := reflect.ValueOf(t).Elem()
	sT := reflect.TypeOf(t).Elem()
	sClone := reflect.ValueOf(tClone).Elem()

	fmt.Println("t=", t)
	fmt.Println("tClone=", tClone)

	for i := 0; i < s.NumField(); i++ {
		fieldName := sT.Field(i).Name
		f2 := sClone.FieldByName(fieldName)
		if !f2.IsValid() {
			continue
		}

		s.Field(i).Set(f2)
	}

	fmt.Println("t=", t)
	fmt.Println("tClone=", tClone)
}

```

https://play.golang.org/p/WTzRYlO9clD